File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-7.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        res=0
        sett = set(s)

        for ch in sett:
            count=l=0
            for r in range(len(s)):
                if s[r]==ch:
                    count+=1
                
                while (r-l+1)-count > k:
                    if s[l] == ch:
                        count-=1
                    l +=1
                res = max(res, r-l+1)
        return res

        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int ans=0;

        for(int ele: nums){
            set.add(ele);
        }        

        for(int ele:nums){
            if(!set.contains(ele-1)){
                int count=0;
                int start=ele;

                while(set.contains(start)){
                    count++;
                    start++;
                }
                ans=Math.max(ans,count);
            }

        }
        return ans;
    }
}
